linker/Linker.Steps/MarkStep.cs

```diff
@@ -1412,6 +1412,9 @@ protected virtual bool AlwaysMarkTypeAsInstantiated (TypeDefinition td)
 				case "Delegate":
 				case "MulticastDelegate":
 					return td.Namespace == "System";
+				// InternalThread doesn't have a ctor that is marked, but it is a very important type not to mess with!
```

Comment:
I don't like this hardcoding of implementation details. If the linker removes it correctly the fix needs to be elsewhere